### PR TITLE
Add Faculty Calendar

### LINKED ISSUE #17: Look into Packages (UI, Calendar, Text Validation)

### PR DESCRIPTION
**Pull request by charkour**
This PR adds multiple configured and customized FullCalendars that represent the faculty schedule. FullCalendar is particular about how imports are sorted which conflicted with our rules. I struggled to find a working alternative so I got fed up and removed most of the import sorting rules. I don't think removing those rules is a huge issue, but I am open to recommendations. 

I ran into one CSS bug I couldn't solve and that was to determine the width of each individual calendar. I found that it is currently 600px wide but I want to make them dynamic rather than static. 

closes #17 

If you have issues running into ESLint, let me know and I'll tell you how to troubleshoot it. Make sure to run npm i in the client folder.

### CODE REVIEW

File: client-course-schedulizer/src/components/Tabs/FacultySchedule/FacultySchedule.scss [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+.calendars-adjacent {
+  width: fit-content;
+  display: flex;
+  flex-direction: row;
+  flex-wrap: nowrap;
+}
+
+.calendar {
+  width: 600px; // This is static, and I don't like it, but I don't know how to make it better
+}
```
Comment (Jdellis818) [inline client-course-schedulizer/src/components/Tabs/FacultySchedule/FacultySchedule.scss]: Would it be possible to make this smaller to fit another professor on the screen?

File: client-course-schedulizer/src/components/Toolbar/ScheduleToolbar/ScheduleToolbar.scss [hunk 2]
```diff
@@ -4,6 +4,16 @@
   justify-content: space-between;
 }
 
+.schedule-toolbar {
+  position: sticky;
```
Comment (Jdellis818) [inline client-course-schedulizer/src/components/Toolbar/ScheduleToolbar/ScheduleToolbar.scss:10]: I like the sticky toolbar!

File: client-course-schedulizer/src/components/reuseables/Calendar/Calendar.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import moment from "moment";
+
+// Plugins
+import interactionPlugin from "@fullcalendar/interaction"; // needed for dayClick
+import timeGridPlugin from "@fullcalendar/timegrid";
+
+import React from "react";
+import "./Calendar.scss";
+
+export const initialDate = "2000-01-02";
```
Comment (Jdellis818) [inline client-course-schedulizer/src/components/reuseables/Calendar/Calendar.tsx:12]: It seems kind of weird to need to set a date when we're not dealing with specific dates. Is there a way to make these events just with a time and not a date? It's fine if there's not.

File: client-course-schedulizer/src/components/reuseables/Calendar/Calendar.tsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        events={events}
+        headerToolbar={false}
+        height="auto"
+        initialDate={initialDate}
+        initialView="timeGridWeek"
+        nowIndicator={false}
+        plugins={[interactionPlugin, timeGridPlugin]}
+        selectable
+        weekends={false}
+      />
```
Comment (Jdellis818) [inline client-course-schedulizer/src/components/reuseables/Calendar/Calendar.tsx]: Could you add `slotMinTime="07:00:00"` and `slotMaxTime="22:00:00"` to limit the hours?

File: client-course-schedulizer/src/components/Tabs/FacultySchedule/FacultySchedule.scss (hunk 1 above)
Comment (Jdellis818) [reply client-course-schedulizer/src/components/Tabs/FacultySchedule/FacultySchedule.scss]: Also, is it possible to make the day labels "M", "T", "W", "Th", and "F"?

**Review by Jdellis818 — CHANGES_REQUESTED** [review]
I really like the package. It seems like it'll likely work well for what we need. I'm a little concerned about the styling limitations, but hopefully we can overcome that. Most of my comments don't require any changes, I was just thinking about little things that aren't too important but could make it better. The only change I really want to see is limiting the hours to 7:00AM-10:00PM. Great work!

File: client-course-schedulizer/src/components/Tabs/FacultySchedule/FacultySchedule.scss (hunk 1 above)
Comment (Jdellis818) [reply client-course-schedulizer/src/components/Tabs/FacultySchedule/FacultySchedule.scss]: `25vw` seems to work for me other than the overlapping day headings. Sometimes it's weird right after I change the scss file, but if I reload the page it's adjusts nicely.

File: client-course-schedulizer/src/components/reuseables/Calendar/Calendar.tsx (hunk 4 above)
Comment (Jdellis818) [reply client-course-schedulizer/src/components/reuseables/Calendar/Calendar.tsx]: Also, is it possible to make the prof headers sticky?

File: client-course-schedulizer/src/components/Tabs/FacultySchedule/FacultySchedule.tsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+export const FacultySchedule = () => {
+  return (
+    <>
+      <ScheduleToolbar />
+      <div className="calendars-wrapper">
+        <div className="calendars-adjacent">
+          {professors.map((prof, index) => {
+            return (
+              <div key={prof} className={`calendar ${index !== 0 ? "hide-axis" : ""}`}>
```
Comment (Jdellis818) [inline client-course-schedulizer/src/components/Tabs/FacultySchedule/FacultySchedule.tsx]: This is good, but it makes the first professor's calendar a bit smaller than the others

File: client-course-schedulizer/src/components/Tabs/FacultySchedule/FacultySchedule.scss (hunk 1 above)
Comment (charkour) [reply client-course-schedulizer/src/components/Tabs/FacultySchedule/FacultySchedule.scss]: Good call, I didn't try reloading the page. It works great! Thanks.

File: client-course-schedulizer/src/components/reuseables/Calendar/Calendar.tsx (hunk 3 above)
Comment (charkour) [reply client-course-schedulizer/src/components/reuseables/Calendar/Calendar.tsx:12]: Unfortunately no. We can just provide functions that hide this nuance when dealing with events so we don't have to worry about it going forward.

File: client-course-schedulizer/src/components/reuseables/Calendar/Calendar.tsx (hunk 4 above)
Comment (charkour) [reply client-course-schedulizer/src/components/reuseables/Calendar/Calendar.tsx]: Yes. It is way too difficult but I managed to do just that.

File: client-course-schedulizer/src/components/Tabs/FacultySchedule/FacultySchedule.tsx (hunk 5 above)
Comment (charkour) [reply client-course-schedulizer/src/components/Tabs/FacultySchedule/FacultySchedule.tsx]: Yeah, I'm not sure the best way to fix this. I'll leave a TODO and recan revisit this later.

**Comment by charkour** [comment]
Getting the Schedule Header to scroll horizontally and still remain sticky to the top of the screen proved to be quite challenging. This may have been one of the most technical challenges I've completed. I think it turned out well. This is kinda bad practice, but I left some todos in this PR that aren't major, but could be addressed in the future when we are polishing the app. Please let me know what you think!

**Review by Jdellis818 — APPROVED** [review]
The spacing of the names are a little weird, but that's definitely an issue we can tackle later on. Great work! LGTM

**Pull request merged**